katacomb/katacomb/qa_report.py

```diff
@@ -60,40 +65,188 @@ def make_pbeam_images(metadata, in_dir, out_dir):
+        make_report_metadata(metadata, os.path.join(qa_dir, qa_report))
+
+        rms_image = pb_filebase + '_aegean_rms'
+        os.rename(os.path.join(pb_dir, rms_image + FITS_EXT),
+                  os.path.join(rms_dir, rms_image + FITS_EXT))
+        _add_missing_axes(os.path.join(rms_dir, rms_image + FITS_EXT))
+        _caption_pngs(rms_dir, rms_image, target, 'RMS PB Corrected')
+
+        bkg_image = pb_filebase + '_aegean_bkg'
+        os.rename(os.path.join(pb_dir, bkg_image + FITS_EXT),
+                  os.path.join(bkg_dir, bkg_image + FITS_EXT))
+        _add_missing_axes(os.path.join(bkg_dir, bkg_image + FITS_EXT))
+        _caption_pngs(bkg_dir, bkg_image, target, 'MEAN PB Corrected')
+
+    make_image_metadata(metadata, '_aegean_rms', rms_dir,
+                        'Continuum PB Corrected RMS Image',
+                        'Continuum PB Corrected RMS image')
+    make_image_metadata(metadata, '_aegean_bkg', bkg_dir,
+                        'Continuum PB Corrected Mean Image',
+                        'Continuum PB Corrected Mean image')
+
+
+def _caption_pngs(in_dir, fits_file, target, label):
+    """Caption and make PNG files"""
+    imghdr = fits.open(os.path.join(in_dir, fits_file + FITS_EXT))[0].header
+    center_freq = imghdr['CRVAL3'] / 1e6
+    caption = f'{target.name} Continuum ({center_freq:.0f} MHz) ({label})'
+    _make_pngs(in_dir, fits_file, caption)
+
+
+def _add_missing_axes(fitsimage):
+    """Aegean strips off the first two axes of fits image, restore them"""
+    image = fits.open(fitsimage)
+    hdr = image[0].header
+    data = image[0].data
+
+    new_hdu = fits.PrimaryHDU(data[np.newaxis, np.newaxis, :, :], hdr)
+    new_hdu.writeto(fitsimage, overwrite=True)
+
+
+def make_image_metadata(metadata, suffix, outdir, rname, desc):
+    """Write a new image metadata file, based on the original pipeline metadata.
+
+    Parameters
+    ----------
+    suffix : str
+        suffix added to the FITS and PNG image file names
+    outdir : str
+        path to write metadata file to
+    rname : str
+        'ReductionName' value in new metadata file
+    desc : str
+        'Description' value in new metdata file
+    """
+    meta_suffix = copy.deepcopy(metadata)
+    filenames = metadata['FITSImageFilename']
+    for i, out_file in enumerate(filenames):
+        out_filebase = os.path.splitext(out_file)[0]
+        out_filebase_suffix = out_filebase + suffix
+        _update_metadata_imagenames(meta_suffix, out_filebase_suffix, i)
+
+    meta_suffix['ProductType']['ReductionName'] = rname
+    desc_prefix = meta_suffix['Description'].split(':')[0]
+    meta_suffix['Description'] = desc_prefix + ': Continuum image PB corrected'
+
+    write_metadata(meta_suffix, outdir)
+
+
+def make_report_metadata(metadata, out_dir):
+    """Write a new reduction product metadata file.
+
+    It is based on the original pipeline metadata, but with
+    updated 'Description' and 'ReductionName' keys.
```

Comment:
Empty line needed above `Parameters`.

Reply:
> Generally looks good!
> 
> I have one style suggestion (maybe just a personal habit smile ): I usually put function/class definitions above the first line that uses them, so when I see e.g. `_caption_pngs(...)` I tend to scroll up to find the definition. The definitions in this file are in no particular order, which means more scrolling for me.

I've made an attempt to re-arrange the functions according to this principle.